### PR TITLE
JBPM-5930 - Update server template id for container by mgmt controlle…

### PR DESCRIPTION
**Pull request by tsurdilo**
…r return code 201

### CODE REVIEW

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java [hunk 1]
```diff
@@ -558,12 +559,71 @@ public void testUpdateContainerConfigOnNotExistingContainer() {
...
+            mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), containerToDeploy);
+            fail("Template key should not be allowed to be changed during update.");
+        } catch (UnexpectedResponseCodeException e) {
+            assertEquals(400, e.getResponseCode());
+            KieServerAssert.assertResultContainsString(e.getMessage(), "Cannot change container template key during update.");
+        }
+
+        assertEquals(2, mgmtControllerClient.listServerTemplates().size());
+        // Check that on other server template is not any container.
+        KieServerAssert.assertNullOrEmpty("Found container in second server template.", mgmtControllerClient.listContainerSpec(secondTemplate.getId()));
```
Comment (jakubschwan) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java]: There would be good to check that container is not changed here

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java [hunk 2]
```diff
@@ -558,12 +559,71 @@ public void testUpdateContainerConfigOnNotExistingContainer() {
...
+    }
+
+    @Test
+    public void testUpdateContainerServerTemplateId() {
```
Comment (jakubschwan) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java]: Please delete this test, there is nothing tested. Probably it is duplication of testTemplateKeyChangeDuringUpdate.

**Comment by jakubschwan** [comment]
Looks good to me. Tests need some minor changes.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java (hunk 1 above)
Comment (tsurdilo) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java]: added. thanks.

**Comment by tsurdilo** [comment]
@jakubschwan amended - thanks! if it's ok now under "Files changed" tab can you use the green "Review changes" dropdown to set your review option.

**Pull request merged**